Sources/MatrixClient/API/Events/RoomEvent.swift

```
@@ -76,8 +76,17 @@ public struct MatrixCodableEvents<Value: Collection>: Codable where Value.Elemen
             wrappedEvent = decoded
         }
 
-        func encode(to _: Encoder) throws {
-            fatalError("Event encoding not implemented.")
+        func encode(to encoder: Encoder) throws {
+            guard let wrappedEvent = wrappedEvent as? Codable,
+                  let contentType = type(of: wrappedEvent) as? MatrixEvent.Type
+            else {
+                return
```

Should this throw an error, or possibly just log that encoding failed? Silent failure could be confusing for someone down the line. (Same comment on `encode(to encoder: Encoder)` below too)

---

Isn't that a problem with the `/sync` endpoint?